server/chat-plugins/datasearch.js

```diff
@@ -1091,36 +1080,53 @@ function runMovesearch(target, cmd, canAll, message) {
 		return {reply: "No search parameters other than 'all' were found. Try '/help movesearch' for more information on this command."};
 	}
 
-	let dex = {};
-	if (targetMons.length) {
-		let intersectionOfLearnsets = [];
-		for (let mon in lsetData) {
-			if (intersectionOfLearnsets.length) {
-				intersectionOfLearnsets = intersectionOfLearnsets.filter(move => {
-					return lsetData[mon][move] !== undefined;
-				});
-			} else {
-				intersectionOfLearnsets = Object.keys(lsetData[mon]);
-			}
-		}
+	const dex = {};
+	for (let move in Dex.data.Movedex) {
```

Comment:
Wow, this seems super inefficient. Pretty confident you should only need the `Move` at the very end for display and should just be operating on `IDs` until then. You're creating 300+ objects with 20-30+ fields here.

Reply:
I think most of Dexsearch is super inefficient. :( My dream is one day to be able to unify the client's teambuilder search and server dexsearch code.

(Actually partially underway! My latest refactor soon-to-be-pushed splits client Dexsearch code away from its UI code.)

Reply:
Changed to work with IDs until the full data of the moves are needed.